server/tests/api_v4/test_enrollment.py

```diff
@@ -17,7 +20,34 @@
         "user",
     ),
 )
-async def test_device_enrollment(invitation: str, coolorg: CoolorgRpcClients):
+@pytest.mark.parametrize(
+    "issue7212",
+    (
+        False,
+        True,
+    ),
+    ids=("normal", "issue7212"),
+)
+async def test_device_enrollment(
+    invitation: str, coolorg: CoolorgRpcClients, issue7212: bool, monkeypatch
+):
+    if issue7212:
+        pg_conduit_listen = PGInviteComponent._conduit_listen
+        memory_conduit_listen = MemoryInviteComponent._conduit_listen
+
+        async def patched_pg_conduit_listen(self, now, ctx, *args, **kwargs):
+            if not ctx.is_greeter and ctx.state == ConduitState.STATE_4_COMMUNICATE:
+                await asyncio.sleep(0.1)
```

Comment:
having a sleep is rather odd here: if claimer operation is slower than the sleep (unlikely but possible in theory), then this wouldn't do anything...

Reply:
> having a sleep is rather odd here: if claimer operation is slower than the sleep (unlikely but possible in theory), then this wouldn't do anything...

The point is merely to reproduce a flaky issue here. During my tests 0.1 second was large enough to reproduce the original problem 100% of the time.

> Could we instead use an event (that greeter would wait on and claimer trigger) ?

Sure I'll find something